### PR TITLE
KT-18674 Join Lines should join strings

### PR DESCRIPTION
**Pull request by t-kameyama**
https://youtrack.jetbrains.com/issue/KT-18674

### CODE REVIEW

File: idea/testData/joinLines/stringTemplate/simple.kt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+fun test1() {
```
Comment (paplorinc) [inline idea/testData/joinLines/stringTemplate/simple.kt]: Some questions:
* does this work for more than 2 lines, e.g. 
```kotlin
val s = "one"
+ ", two"
+ ", three"
```
* does it work for non-strings, e.g.
```java
val s = "x=" +
0
```

File: idea/testData/joinLines/stringTemplate/simple.kt.after [hunk 2]
```diff
@@ -0,0 +1,3 @@
+fun test1() {
+    val s = <caret>"foobar"
```
Comment (paplorinc) [inline idea/testData/joinLines/stringTemplate/simple.kt.after]: should the caret maybe placed to the same place, i.e.
```kotlin
"foo<caret>bar"
```

Comment (yole) [reply idea/testData/joinLines/stringTemplate/simple.kt.after]: Yes, the caret needs to be in the same place.

File: idea/testData/joinLines/stringTemplate/simple2.kt.after [hunk 3]
```diff
@@ -0,0 +1,4 @@
+fun test2() {
+    val foo = "1"
+    val s = <caret>"${foo}bar"
```
Comment (yole) [inline idea/testData/joinLines/stringTemplate/simple2.kt.after]: I think this is surprising behavior. Lines should be joined into a single expression only if both the left hand side and the right hand side of the concatenation are constants.

File: idea/testData/joinLines/stringTemplate/simple.kt (hunk 1 above)
Comment (yole) [reply idea/testData/joinLines/stringTemplate/simple.kt]: The Join Lines action joins the line where the caret is with the next line. It always acts on exactly two lines.

Comment (paplorinc) [reply idea/testData/joinLines/stringTemplate/simple.kt]: Thanks

File: idea/testData/joinLines/stringTemplate/simple.kt.after (hunk 2 above)
Comment (t-kameyama) [reply idea/testData/joinLines/stringTemplate/simple.kt.after]: Fixed

File: idea/testData/joinLines/stringTemplate/simple2.kt.after (hunk 3 above)
Comment (t-kameyama) [reply idea/testData/joinLines/stringTemplate/simple2.kt.after]: Fixed

File: idea/testData/joinLines/stringTemplate/simple.kt (hunk 1 above)
Comment (t-kameyama) [reply idea/testData/joinLines/stringTemplate/simple.kt]: Fixed

**Comment by yole** [comment]
Thanks a lot for your contribution, and sorry for the delay with the review!

**Pull request merged**